app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepcustomize/SelectCoverageViewModel.kt

```diff
@@ -0,0 +1,328 @@
+  private val params: InsuranceCustomizationParameters,
+  private val tierRepository: ChangeTierRepository,
+  val getCurrentContractDataUseCase: GetCurrentContractDataUseCase,
+) : MoleculePresenter<SelectCoverageEvent, SelectCoverageState> {
+  @Composable
+  override fun MoleculePresenterScope<SelectCoverageEvent>.present(
+    lastState: SelectCoverageState,
+  ): SelectCoverageState {
+    var chosenTier by remember { mutableStateOf(if (lastState is Success) lastState.uiState.chosenTier else null) }
+    var chosenQuote by remember { mutableStateOf(if (lastState is Success) lastState.uiState.chosenQuote else null) }
+    var chosenTierInDialog by remember {
+      mutableStateOf(if (lastState is Success) lastState.uiState.chosenTier else null)
+    }
+    var chosenQuoteInDialog by remember {
+      mutableStateOf(if (lastState is Success) lastState.uiState.chosenQuote else null)
+    }
+    var quoteToNavigateFurther by remember { mutableStateOf<TierDeductibleQuote?>(null) }
+    var quotesToCompare by remember { mutableStateOf<List<TierDeductibleQuote>?>(null) }
+
+    var currentPartialState by remember { mutableStateOf(mapLastStateToPartial(state = lastState)) }
+
+    var currentContractLoadIteration by remember { mutableIntStateOf(0) }
+
+    CollectEvents { event ->
+      when (event) {
+        is ChangeDeductibleForChosenTier -> {
+          chosenQuote = chosenQuoteInDialog
+        }
+
+        is ChangeTier -> {
+          val state = currentPartialState
+          if (state !is PartialUiState.Success) return@CollectEvents
+          // set newly chosen tier
+          val locallyChosen = chosenTierInDialog
+          locallyChosen?.let { local ->
+            chosenTier = local
+            // try to pre-choose a quote with the same deductible and newly chosen coverage
+            // if there is no such quote, the deductible will not be per-chosen
+            val previouslyChosenDeductible = chosenQuote?.deductible
+            val quoteWithNewTierOldDeductible =
+              state.map[local]!!.firstOrNull { it.deductible == previouslyChosenDeductible }
+            chosenQuote = quoteWithNewTierOldDeductible
+          }
+        }
+
+        ClearNavigationStep -> {
+          quoteToNavigateFurther = null
+          quotesToCompare = null
+        }
+
+        SubmitChosenQuoteToContinue -> {
+          val state = currentPartialState
+          if (state !is PartialUiState.Success) return@CollectEvents
+          if (chosenQuote != state.currentActiveQuote) {
+            quoteToNavigateFurther = chosenQuote
+          }
+        }
+
+        Reload -> currentContractLoadIteration++
+
+        LaunchComparison -> {
+          if (currentPartialState !is PartialUiState.Success) return@CollectEvents
+          val filtered = (currentPartialState as PartialUiState.Success).map.values.flatten()
+            .filter { it.deductible == chosenQuote?.deductible }
+          quotesToCompare =
+            filtered
```

Comment:
so this I think will be working differently. the comparison API is not here yet, and what I think will be the input for that query will be termsVersionsIds. [link](https://hedviginsurance.slack.com/archives/C07MM6F0DK2/p1726217330185079) So this is just a temporary stub here. (related - changed the SixTabs impl so it would actually show one tab as well)